app/api/csv.py

```diff
@@ -38,6 +38,9 @@ def make_csv_response(columns, data):
     si = StringIO()
     writer = csv.writer(si)
 
+    if not columns:
+        []
```

Comment:
this looks a bit odd - do you ever anticipate having an empty list of columns to this, and if so, what do you want to return?

Reply:
ehhh I'm not sure about this. I took the previous PR and merged on top of master after all the other changes were committed, so maybe it's a weird artifact.